Sources/AsyncAlgorithms/Locking.swift

```diff
@@ -13,14 +13,18 @@
 import Darwin
 #elseif canImport(Glibc)
 import Glibc
+#elseif canImport(Musl)
+import Musl
 #elseif canImport(WinSDK)
 import WinSDK
+#else
+#error("Unsupported platform")
 #endif
 
 internal struct Lock {
 #if canImport(Darwin)
   typealias Primitive = os_unfair_lock
-#elseif canImport(Glibc)
+#elseif canImport(Glibc) || canImport(Musl)
   typealias Primitive = pthread_mutex_t
 #elseif canImport(WinSDK)
   typealias Primitive = SRWLOCK
```

Comment:
Can we make the `#else` case fail with an `#error` to avoid running into this again

Reply:
I did add it at the top, but can add here as well